### PR TITLE
[BEAM-7641] Collect xunit statistics for Py ITs

### PR DESCRIPTION
**Pull request by udim**
Affects only ITs run using run_integration_test.sh.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by udim** [comment]
run python postcommit

**Comment by udim** [comment]
run seed job

**Comment by udim** [comment]
Run Python PostCommit

**Comment by udim** [comment]
Run Python Dataflow ValidatesContainer

**Comment by udim** [comment]
Run Python Dataflow ValidatesRunner

**Comment by udim** [comment]
Run Python Spark ValidatesRunner

**Comment by udim** [comment]
run python precommit

**Comment by udim** [comment]
run seed job

**Comment by udim** [comment]
run seed job

**Comment by udim** [comment]
Run Python PostCommit

**Comment by udim** [comment]
Run Python Dataflow ValidatesContainer

**Comment by udim** [comment]
Run Python Dataflow ValidatesRunner

**Comment by udim** [comment]
run python precommit

**Comment by udim** [comment]
Run Python PostCommit

**Comment by udim** [comment]
R: @lukecwik @markflyhigh 
I know there are a lot of failures, but I assert that:
1. none of them are caused by this PR;
2. this PR will help us identify test failure frequency (for ITs).

**Comment by udim** [comment]
run python precommit

**Comment by udim** [comment]
run python postcommit

File: sdks/python/scripts/run_integration_test.sh [hunk 1]
```diff
@@ -48,6 +48,8 @@
 #                      during execution. Commonly used options like `--attr`,
 #                      `--tests`, `--nologcapture`. More can be found in
 #                      https://nose.readthedocs.io/en/latest/man.html#options
+#     suite         -> Namespace for this run of tests. Required if running
+#                      under Jenkins.
```
Comment (markflyhigh) [inline sdks/python/scripts/run_integration_test.sh]: Can you explain more about it? It's still not clear to me what string I should give when adding new test and how it's been used from nose. Can it be an arbitrary string? Should it be unique?

File: sdks/python/scripts/run_integration_test.sh [hunk 2]
```diff
@@ -226,6 +240,9 @@ fi
 
 echo ">>> RUNNING integration tests with pipeline options: $PIPELINE_OPTS"
 echo ">>>   test options: $TEST_OPTS"
+# TODO(udim): Pass $SUITE once migrated to pytest. xunitmp doesn't support suite
+#   names.
```
Comment (markflyhigh) [inline sdks/python/scripts/run_integration_test.sh]: Can you create a jira to track it? And maybe link to pytest epic? it's easy to forget after a while.

File: sdks/python/scripts/run_integration_test.sh [hunk 3]
```diff
@@ -48,6 +48,8 @@
 #                      during execution. Commonly used options like `--attr`,
 #                      `--tests`, `--nologcapture`. More can be found in
 #                      https://nose.readthedocs.io/en/latest/man.html#options
```
Comment (markflyhigh) [inline sdks/python/scripts/run_integration_test.sh]: `test_opts` is defined to hold Nose configs. Do you have particular reason to define it separately?

Comment (udim) [reply sdks/python/scripts/run_integration_test.sh]: Yes, it is also used in the xml output filename.

File: sdks/python/scripts/run_integration_test.sh (hunk 1 above)
Comment (udim) [reply sdks/python/scripts/run_integration_test.sh]: It's used to differentiate running the same tests with different arguments/python versions/etc.

File: sdks/python/scripts/run_integration_test.sh (hunk 2 above)
Comment (udim) [reply sdks/python/scripts/run_integration_test.sh]: Added existing JIRA

File: sdks/python/scripts/run_integration_test.sh (hunk 1 above)
Comment (udim) [reply sdks/python/scripts/run_integration_test.sh]: Added more explanation to the comment.

**Comment by udim** [comment]
run python postcommit

**Comment by udim** [comment]
run python postcommit

**Comment by udim** [comment]
run python postcommit

**Comment by udim** [comment]
R: @tvalentyn could you take a look?

**Comment by tvalentyn** [comment]
LGTM

**Pull request merged**